base/wheels/Dockerfile

```diff
@@ -43,35 +48,41 @@ RUN CFLAGS="-DACCEPT_USE_OF_DEPRECATED_PROJ_API_H=1" \
   --no-binary=:all: \
   --no-build-isolation \
   --no-deps \
+  -c /wheels/constraints.txt \
   cartopy
 
 RUN pip3 wheel \
   --no-cache-dir \
   --no-binary=:all: \
   --no-build-isolation \
   --no-deps \
+  -c /wheels/constraints.txt \
   fiona
 
 RUN pip3 wheel \
   --no-cache-dir \
   --no-binary=:all: \
   --no-build-isolation \
   --no-deps \
+  -c /wheels/constraints.txt \
   shapely
 
 RUN pip3 wheel \
   --no-cache-dir \
   --no-binary=:all: \
   --no-build-isolation \
   --no-deps \
-  'pyproj==2.6.*'
+  -c /wheels/constraints.txt \
+  pyproj
 
 RUN pip3 wheel \
   --no-cache-dir \
   --no-binary=:all: \
   --no-build-isolation \
   --no-deps \
-  rasterio==1.1.6
+  -c /wheels/constraints.txt \
+  rasterio
 
+RUN rm constraints.txt
```

Comment:
this step is not actually reducing docker image size.

Reply:
Good point. It only adds another layer (which is slightly undesirable)